### ISSUE TITLE
[DevTools Bug]: Using different React instances across multiple frames throws errors

### ISSUE THREAD

**Issue by Mesoptier** [facebook/react#26797]
### Website or app

https://codepen.io/mesoptier/pen/qBJopod

### Repro steps

1. Visit the codepen linked above.
2. Be sure to open the Debug View, so codepen doesn't add any additional iframes ([more info](https://blog.codepen.io/documentation/debug-view/)).
3. Observe:
   - Warning in the Console tab: `Invalid renderer id "1"`.
   - Error in the Components tab: `Uncaught Error: Cannot add node "3" because a node with that id is already in the Store`.
   - After dismissing the error, all components seem to be accounted for (you might need to adjust the "Hide components where..." option).

In my actual application the error pops up for every change in the tree, making the devtools virtually unusable.

Note: This only seems to happen if the iframe is added some time after the initial React tree was mounted in the parent window. When I remove the `setTimeout` so the iframe is added synchronously, I no longer see any warnings.

Related issues:
- https://github.com/facebook/react/issues/26787
- https://github.com/facebook/react/issues/26793

### How often does this bug happen?

Every time

### DevTools version ~(automated)~ (manual)

4.27.7 (5/7/2023) in Google Chrome Version 112.0.5615.165 (Official Build) (64-bit)

### Error message ~(automated)~ (manual)

Cannot add node "3" because a node with that id is already in the Store.

### Error call stack ~(automated)~ (manual)

```
at chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/main.js:29031:41
    at bridge_Bridge.emit (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/main.js:27054:22)
    at chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/main.js:27223:14
    at listener (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/main.js:57493:39)
```

**Comment by hoxyq** [comment]
Hey @Mesoptier, thanks for reporting this.

Can confirm this issue, but I am getting different error on my side and only when refreshing the page. Extension works as expected for me when I open the page for the first time.

I will put up a fix for this.

**Comment by hoxyq** [comment]
@Mesoptier can you please also validate this on your side?

You can download a build from our CircleCI pipeline here - https://app.circleci.com/pipelines/github/facebook/react/42301/workflows/8a3dd19a-f9f0-4af2-bd91-b030356bbf98/jobs/684003/artifacts

Chrome extension should be in `chrome-extension` archive, you can then install it via `Load unpacked` dialog in `chrome://extensions/`

**Comment by Mesoptier** [comment]
@hoxyq Yes! I'm no longer seeing the warning/errors when I use the build you linked. Thanks for the fix :)

**Comment by hoxyq** [comment]
> @hoxyq Yes! I'm no longer seeing the warning/errors when I use the build you linked. Thanks for the fix :)

Great, thanks for helping resolving this. This will be included in v4.27.8 release, which I will publish this week.
